ukvm/ukvm_module_net.c

```diff
@@ -178,8 +178,10 @@ static void hypercall_netwrite(struct ukvm_hv *hv, ukvm_gpa_t gpa)
     int ret;
 
     ret = write(netfd, UKVM_CHECKED_GPA_P(hv, wr->data, wr->len), wr->len);
-    assert(wr->len == ret);
-    wr->ret = 0;
+    if (wr->len == ret)
+        wr->ret = 0;
+    else
+        wr->ret = -1;
 }
```

Comment:
This is problematic since host-side errors can also be permanent (eg. "Interface is down"). What specific error were you getting, and under what conditions?